src/TestFramework/Coverage/XmlReport/JUnitTestCaseSorter.php

```diff
@@ -49,10 +49,20 @@ final class JUnitTestCaseSorter
     /**
      * @param CoverageLineData[] $coverageTestCases
      *
-     * @return string[]
+     * @return iterable<string>
      */
-    public function getUniqueSortedFileNames(array $coverageTestCases): array
+    public function getUniqueSortedFileNames(array $coverageTestCases): iterable
```

Comment:
After taking another look, this `iterable` case does not make sense to me. There is two paths:

- yielding only one result
- yield multiple results

In the second, the yields are preceded by a `usort()` on an array. So there is really no benefits from yielding here from a performance or memory PoV.

Looking at the parent `MutationConfigBuilder`, it also iterates over all the results right away and it does not appear it can take advantage of an iterator either

Reply:
How about now?

The point is `MutationConfigBuilder` doesn't need an array.

Reply:
>The point is MutationConfigBuilder doesn't need an array.

Sure, it just needs an iterable, but array is always much simpler to deal with than `iterable` or `Generator`, so if the sub-type is enough why expanding it? It's neither a case where there would be a performance benefit or make a more friendly/robust API